File: src/main/java/com/mudhut/nudge/users/repositories/UserRepository.java
package com.mudhut.nudge.users.repositories;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.mudhut.nudge.users.models.User;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    boolean existsByEmail(String email);

    boolean existsByPhoneNumber(String phoneNumber);

    Optional<User> findByEmail(String email);
}

File: src/main/kotlin/com/mudhut/nudge/users/repositories/UserRepository.kt
package com.mudhut.nudge.users.repositories

import com.mudhut.nudge.users.entities.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun existsByEmail(email: String): Boolean
    fun existsByPhoneNumber(phoneNumber: String): Boolean
    fun findByEmail(email: String): Optional<User>
}
